nestkernel/conn_builder.cpp

```diff
@@ -74,75 +71,145 @@ nest::ConnBuilder::ConnBuilder( NodeCollectionPTR sources,
   updateValue< bool >( conn_spec, names::make_symmetric, make_symmetric_ );
 
   // read out synapse-related parameters ----------------------
-  if ( not syn_spec->known( names::synapse_model ) )
-  {
-    throw BadProperty( "Synapse spec must contain synapse model." );
-  }
-  const std::string syn_name = ( *syn_spec )[ names::synapse_model ];
-  if ( not kernel().model_manager.get_synapsedict()->known( syn_name ) )
+
+  // synapse-specific parameters
+  std::set< Name > skip_set = {
+    names::weight, names::delay, names::min_delay, names::max_delay, names::num_connections, names::synapse_model
+  };
+
+  default_weight_.resize( syn_spec.size() );
+  default_delay_.resize( syn_spec.size() );
+  default_weight_and_delay_.resize( syn_spec.size() );
+  weight_.resize( syn_spec.size() );
+  delay_.resize( syn_spec.size() );
+  synapse_params_.resize( syn_spec.size() );
+  synapse_model_id_.resize( syn_spec.size() );
+  synapse_model_id_[ 0 ] = kernel().model_manager.get_synapsedict()->lookup( "static_synapse" );
+  param_dicts_.resize( syn_spec.size() );
+
+  int indx = 0;
+  for ( auto syn_params : syn_spec )
   {
-    throw UnknownSynapseType( syn_name );
-  }
+    if ( not syn_params->known( names::synapse_model ) )
+    {
+      throw BadProperty( "Synapse spec must contain synapse model." );
+    }
+    const std::string syn_name = ( *syn_params )[ names::synapse_model ];
+    if ( not kernel().model_manager.get_synapsedict()->known( syn_name ) )
+    {
+      throw UnknownSynapseType( syn_name );
+    }
 
-  synapse_model_id_ = kernel().model_manager.get_synapsedict()->lookup( syn_name );
+    index synapse_model_id = kernel().model_manager.get_synapsedict()->lookup( syn_name );
+    synapse_model_id_[ indx ] = synapse_model_id;
 
-  // We need to make sure that Connect can process all synapse parameters
-  // specified.
-  const ConnectorModel& synapse_model = kernel().model_manager.get_synapse_prototype( synapse_model_id_, 0 );
-  synapse_model.check_synapse_params( syn_spec );
+    // We need to make sure that Connect can process all synapse parameters
+    // specified.
+    const ConnectorModel& synapse_model = kernel().model_manager.get_synapse_prototype( synapse_model_id, 0 );
```

Comment:
What is the second argument to `get_syn_proto()`? Actually, we have
```c++
ConnectorModel& get_synapse_prototype( synindex syn_id, thread t = 0 );
```
so we could drop the 2nd arg here and reduce confusion.